services/relay.c

```diff
@@ -0,0 +1,66 @@
+#include "jd_services.h"
+#include "interfaces/jd_pins.h"
+#include "jacdac/dist/c/relay.h"
+
+struct srv_state {
+    SRV_COMMON;
+    uint8_t intensity;
+    uint8_t variant;
+    uint32_t max_switching_current;
+    relay_params_t params;
```

Comment:
@mmoskal  I could probably move the registers (except intensity) into params. I had such a silly battle to actually program the G030's though, and I can't debug them whilst they're running the app (JD_DATA is muxed with SWCLK). Development is a true pain with these things.

Reply:
I think this is no big deal